toolkit/components/uniffi-bindgen-gecko-js/src/templates/js/Object.jsm

```diff
@@ -0,0 +1,121 @@
+     *      to a newly constructed {{ object.nm() }}
+     */
+    static init({{cons.arg_names()}}) {
+        const liftResult = (resultPtr) => {
+            return new {{ object.nm() }}(resultPtr);
+        };
+        {%- match cons.throws_type() %}
+        {%- when Some with (err_type) %}
+        const liftError = (data) => {{ err_type.ffi_converter() }}.lift(data);
+        {%- else %}
+        const liftError = null;
+        {%- endmatch %}
+    
+        const callResult = {{ ci.scaffolding_name() }}.{{cons.ffi_func().nm()}}(
+            {%- for arg in cons.arguments() -%}
+            {{ arg.lower_fn_name() }}({{ arg.name() }}),
+            {%- endfor %}
+        )
+        {%- if cons.is_async() %}
+        return callResult.then((result) => handleRustResult(result,  liftResult, liftError));
+        {%- else %}
+        return handleRustResult(callResult,  liftResult, liftError);
+        {%- endif %}
+    }
+    {%- else %}
+    
+    /**
+     * A constructor for {{ object.nm() }}.
+     * 
+     * @returns { {{ object.nm() }} }
+     */
+    static init({{ cons.arg_names() }}) {
+        const liftResult = (resultPtr) => {
+            return new {{ object.nm() }}(resultPtr);
+        };
+        {%- match cons.throws_type() %}
+        {%- when Some with (err_type) %}
+        const liftError = (data) => {{ err_type.ffi_converter() }}.lift(data);
+        {%- else %}
+        const liftError = null;
+        {%- endmatch %}
+    
+        const callResult = {{ ci.scaffolding_name() }}.{{cons.ffi_func().nm()}}(
+            {%- for arg in cons.arguments() -%}
+            {{ arg.lower_fn_name() }}({{ arg.name() }}),
+            {%- endfor %}
+        )
+        return handleRustResult(callResult,  liftResult, liftError);
+    }
+    {%- endif %}
+    {%- else %}
+    {%- endmatch %}
+
+    {%- for meth in object.methods() %}
+    {{ meth.nm() }}({{ meth.arg_names() }}) {
+        const liftResult = (result) => {{ meth.ffi_return_type() }}.lift(result);
+        {%- match meth.throws_type() %}
+        {%- when Some with (err_type) %}
+        const liftError = (data) => {{ err_type.ffi_converter() }}.lift(data);
+        {%- else %}
+        const liftError = null;
+        {%- endmatch %}
+    
+        const callResult = {{ ci.scaffolding_name() }}.{{meth.ffi_func().nm()}}(this.ptr,
+            {%- for arg in meth.arguments() -%}
+            {{ arg.lower_fn_name() }}({{ arg.name() }}),
+            {%- endfor %}
+        )
+        {%- if meth.is_async() %}
+        return callResult.then((result) => handleRustResult(result,  liftResult, liftError));
+        {%- else %}
+        return handleRustResult(callResult,  liftResult, liftError);
+        {%- endif %}
+    }
+    {%- endfor %}
+}
+
+class {{ ffi_converter }} {
+    static lift(value) {
+        return new {{ object.nm() }}(value);
+    }
+
+    static lower(value) {
+        return value.ptr;
```

Comment:
haven't tested this yet, just wanted to point it out. The fixtures we have so far in this repo don't cover lowering an interface